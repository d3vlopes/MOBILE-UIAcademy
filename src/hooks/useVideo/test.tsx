import React from 'react'
import { renderHook } from '@testing-library/react-hooks'
import { act } from 'react-test-renderer'

import { useVideo, VideoProvider, VideoProviderProps } from '.'

describe('useVideo', () => {
  it('should change the value of isFullScreen', () => {
    const wrapper = ({ children }: VideoProviderProps) => (
      <VideoProvider>{children}</VideoProvider>
    )

    const { result } = renderHook(() => useVideo(), {
      wrapper,
    })

    act(() => {
      result.current.changeFullScreenValue()
    })

    expect(result.current.isFullScreen).toBe(true)
  })

  it('should exec onPrevious function', () => {
    const wrapper = ({ children }: VideoProviderProps) => (
      <VideoProvider>{children}</VideoProvider>
    )

    const { result } = renderHook(() => useVideo(), {
      wrapper,
    })

    act(() => {
      result.current.onPrevious()
    })

    // expect()
  })

  it('should exec onNext function', () => {
    const wrapper = ({ children }: VideoProviderProps) => (
      <VideoProvider>{children}</VideoProvider>
    )

    const { result } = renderHook(() => useVideo(), {
      wrapper,
    })

    act(() => {
      result.current.onNext()
    })

    // expect()
  })
})
