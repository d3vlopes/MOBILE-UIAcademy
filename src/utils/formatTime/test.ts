import { formatTime } from '.'

describe('formatTime', () => {
  it('should return the minutes formatted', () => {
    const time = 45

    expect(formatTime(time)).toEqual('45min')
  })

  it('should return the hours formatted', () => {
    const time = 80

    expect(formatTime(time)).toEqual('1hr 20min')
  })
})
