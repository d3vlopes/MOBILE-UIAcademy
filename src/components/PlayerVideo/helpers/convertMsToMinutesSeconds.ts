import { padTo2Digits } from '../../../utils/formatTime/padTo2Digits'

export function convertMsToMinutesSeconds(milliseconds: number) {
  let minutes = Math.floor(milliseconds / 60000)
  let seconds = Math.floor((milliseconds % 60000) / 1000)

  if (Number.isNaN(minutes) || Number.isNaN(seconds)) {
    minutes = 0
    seconds = 0
  }

  return seconds === 60
    ? `${minutes + 1}:00`
    : `${minutes}:${padTo2Digits(seconds, 2)}`
}
