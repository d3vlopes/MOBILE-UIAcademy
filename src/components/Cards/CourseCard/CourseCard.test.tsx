import React from 'react'

import { renderWithTheme } from '../../../utils/test/helpers'

import FigmaSvg from '../../assets/figma.svg'

import { CourseCard } from '.'

describe('<CourseCard />', () => {
  it('should render by default', () => {
    const { getByTestId } = renderWithTheme(
      <CourseCard
        testID="courseCard"
        svg={FigmaSvg}
        title="Figma"
        amountClasses="7 aulas"
        amountHours="4h"
      />,
    )

    const courseCard = getByTestId('courseCard')
    const wrapperIcon = getByTestId('wrapperIcon')
    const titleCourse = getByTestId('titleCourse')
    const detailsCourse = getByTestId('detailsCourse')

    expect(courseCard).toBeTruthy()
    expect(wrapperIcon).toBeTruthy()
    expect(titleCourse).toBeTruthy()
    expect(detailsCourse).toBeTruthy()
  })
  it('should match snapshot', () => {
    const { toJSON } = renderWithTheme(
      <CourseCard
        testID="courseCard"
        svg={FigmaSvg}
        title="Figma"
        amountClasses="7 aulas"
        amountHours="4h"
      />,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
