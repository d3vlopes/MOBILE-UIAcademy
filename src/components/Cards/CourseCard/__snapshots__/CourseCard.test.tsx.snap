// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseCard /> should match snapshot 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.105}
  collapsable={false}
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  style={
    Object {
      "backgroundColor": "#0B0B0B",
      "borderRadius": 18,
      "height": 263,
      "justifyContent": "center",
      "minWidth": 271,
      "paddingBottom": 47,
      "paddingLeft": 47,
      "paddingRight": 47,
      "paddingTop": 47,
    }
  }
  testID="courseCard"
  underlayColor="black"
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderRadius": 18,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#343434",
          "borderRadius": 12,
          "height": 75,
          "justifyContent": "center",
          "width": 75,
        },
      ]
    }
    testID="wrapperIcon"
  >
    <SvgMock />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#F6F6FA",
          "fontFamily": "Epilogue_600SemiBold",
          "fontSize": 16,
          "marginTop": 8,
        },
      ]
    }
    testID="titleCourse"
  >
    Figma
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#F6F6FA",
            "fontFamily": "Epilogue_400Regular",
            "fontSize": 14,
            "marginTop": 4,
            "opacity": 0.7,
          },
        ]
      }
      testID="detailsCourse"
    >
      7 aulas
       - 
      4h
    </Text>
  </View>
</RNGestureHandlerButton>
`;
